### PR TITLE
Spec v2/adding ttls to UI

### PR DESCRIPTION
**Pull request by chakri642**
## Type of Change

- [ ] Bugfix
- [x] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates

### Additional Changes

- [ ] This PR modifies the database schema (database migration added)
- [ ] This PR modifies dhall configs/environment variables
- [ ] This PR contains API breaking changes

## Checklist

- [ ] I formatted the code and addressed linter errors `./dev/format-all-files.sh`
- [ ] I reviewed submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable
- [ ] No leak detected in leakcanary

### CODE REVIEW

File: Backend/app/rider-platform/rider-app/Main/src/API/UI/Select.hs [hunk 1]
```diff
@@ -73,15 +81,34 @@ type API =
...
-  select2
+  select
+    :<|> select2
     :<|> selectList
     :<|> selectResult
     :<|> cancelSearch
 
+select :: (Id DPerson.Person, Id Merchant.Merchant) -> Id DEstimate.Estimate -> DSelect.DSelectReq -> FlowHandler DSelect.DSelectResultRes
+select (personId, _) estimateId req = withFlowHandlerAPI . withPersonIdLogTag personId $ do
+  dSelectReq <- DSelect.select personId estimateId req
```
Comment (NikithShetty) [inline Backend/app/rider-platform/rider-app/Main/src/API/UI/Select.hs:92]: This statement should be inside Redis Lock

File: Backend/app/rider-platform/rider-app/Main/src/Domain/Action/UI/Select.hs [hunk 2]
```diff
@@ -121,7 +129,14 @@ instance FromJSON CancelAPIResponse where
   parseJSON err = typeMismatch "Object APISuccess" err
 
 select :: Id DPerson.Person -> Id DEstimate.Estimate -> DSelectReq -> Flow DSelectRes
-select personId estimateId req@DSelectReq {..} = do
+select personId estimateId req = do
+  now <- getCurrentTime
+  selectRes <- select2 personId estimateId req
+  when (selectRes.estimate.validTill < now) $ throwError (InvalidRequest $ "Estimate expired " <> show selectRes.estimate.id) -- select validation check
```
Comment (NikithShetty) [inline Backend/app/rider-platform/rider-app/Main/src/Domain/Action/UI/Select.hs:135]: This check should come first, before `selectRes <- select2 personId estimateId req`

File: Backend/app/rider-platform/rider-app/Main/src/API/UI/Confirm.hs [hunk 3]
```diff
@@ -68,12 +71,18 @@ confirm (personId, _) quoteId mbPaymentMethodId =
   withFlowHandlerAPI . withPersonIdLogTag personId $ do
     dConfirmRes <- DConfirm.confirm personId quoteId mbPaymentMethodId
     becknInitReq <- ACL.buildInitReqV2 dConfirmRes
```
[the remark this answers is not in the document]
Comment (chakri642) [reply Backend/app/rider-platform/rider-app/Main/src/API/UI/Confirm.hs:73]: there is some other place in code where `buildInitReqV2` is called, again we should query there so we can't send `bapconfig` as param.

File: Backend/app/rider-platform/rider-app/Main/src/API/UI/Select.hs (hunk 1 above)
Comment (chakri642) [reply Backend/app/rider-platform/rider-app/Main/src/API/UI/Select.hs:92]: but i want to use a parameter `autoAssignEnabled` which is in `dSelectReq`. so if i keep it in redis lock i can't use that outside.

**Review by NikithShetty — APPROVED** [review]

**Pull request merged**